pkg/systemd/quadlet/unitdirs.go

```diff
@@ -0,0 +1,248 @@
+package quadlet
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/user"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/containers/podman/v5/pkg/logiface"
+)
+
+// This returns whether a file has an extension recognized as a valid Quadlet unit type.
+func IsExtSupported(filename string) bool {
+	ext := filepath.Ext(filename)
+	_, ok := SupportedExtensions[ext]
+	return ok
+}
+
+// This returns default install paths for .Quadlet files for `rootless` and `root` user.
+// Defaults to `/etc/containers/systemd` for root user and `$XDG_CONFIG_HOME/containers/systemd`
+// for rootless users.
+func GetInstallUnitDirPath(rootless bool) string {
+	if rootless {
+		configDir, err := os.UserConfigDir()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Warning: %v", err)
```

Comment:
this should not print directly to stderr. if we have the logiface then just use this.